### PR TITLE
Pre-commit Setup

### PR DESCRIPTION
**Pull request by naqeeb**
### Description
Pre-commit config to run 3 checks [end of file, trailing whitespace and black (.py files only)]

### Implementation Notes
1. This will only run on files that were MODIFIED during the specific commit
2. This implementation requires a developer to opt-in: They would need to set up the pre-commit hook in their development environment 
3. This pre-commit hook can be bypassed by using `--no-verify` option during the commit
4. This will lay the groundwork for future CI/CD work where we can strictly enforce formatting

### Testing
Created a test branch and added a new line to democracylab/admin.py to trigger the checks. Please note that the pre-commit checks only run on MODIFIED files.

#### Pre-commit Checks
Once the change is staged, you can trigger the commit via git command line or Sourcetree
Screen Shot 2022-09-27 at 1 30 42 PM

#### Python Black formatting changes
Screen Shot 2022-09-27 at 1 31 03 PM

### Additional Information
- Updated instructions in the Contributor guide: https://docs.google.com/document/d/1OLQPFFJ8oz_BxpuxRxKKdZ2brmlUkVN3ICTdbA_axxY/edit?pli=1#
- Github Ticket: https://github.com/DemocracyLab/CivicTechExchange/issues/905

### CODE REVIEW

**Review by PeterBreen — COMMENTED** [review]
I tested this branch this morning (this feature is going to be great, very excited) and I did get it to work successfully after resolving one problem.

I'm testing on a pretty old Ubuntu VM, where the default python is 3.6.9 but the minimum for this to work is 3.7; otherwise it fails with  `SyntaxError: future feature annotations is not defined`. 

I resolved that adding a default version of Python to .pre-commit-config.yaml:
``` 
default_language_version:
    python: python3.7
```
After that, it worked flawlessly. I don't know how many people are using old and weird dev VMs for dlab (probably one, me) but it might be worth an update to the .yaml or the documentation in the contributor's guide?

**Comment by naqeeb** [comment]
@PeterBreen - I've made the change. You can review it once more and approve if it looks good to you.

**Review by marlonkeating — APPROVED** [review]
Tested on Linux and Windows.  Works great!

**Pull request merged**
